Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -567,9 +567,6 @@ class TabTrayController: UIViewController {
         // If we are exiting private mode and we have the close private tabs option selected, make sure
         // we clear out all of the private tabs
         let exitingPrivateMode = !privateMode && profile.prefs.boolForKey("settings.closePrivateTabs") ?? false
-        if exitingPrivateMode {
-            tabManager.removeAllPrivateTabsAndNotify(false)
-        }
```

Comment:
Good catch! I think switching via tabtray is one of the only places where a new tab isnt selected when exiting.